### DISCUSSION TITLE
How do i modify query parameters ?

### DISCUSSION THREAD

**Discussion by YunigeCDilip** [spatie/laravel-query-builder#728] [General]
**Instead of using**
`users?filter[name]=Joe&filter[position]=developer`

**can we make it to:**
`users?name=Joe&position=developer`

**Comment by dominikb** [comment]
I think this behavior can not be introduced in the package by default.

If you really want you could, however, add a custom middleware that automatically adds those _'unknown'_ parameters as filters:

```php
// A middleware that achieves this behavior (may have unintended consequences!)
class UseAllParamsAsDefaultFilters
{
    public function handle(Request $request, Closure $next)
    {
        $request->query->set('filter', $request->query->all());
        return $next($request);
    }
}

// Now ?name=dominikb works just like ?filter[name]=dominikb
```
